nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/pubsub/AbstractGCPubSubProcessor.java

```diff
@@ -58,6 +73,19 @@ public abstract class AbstractGCPubSubProcessor extends AbstractGCPProcessor imp
 
     private static final Set<Relationship> relationships = Collections.unmodifiableSet(new HashSet<>(Arrays.asList(REL_SUCCESS, REL_FAILURE)));
 
+    @Override
+    public List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return Collections.unmodifiableList(Arrays.asList(
+                PROJECT_ID,
+                GCP_CREDENTIALS_PROVIDER_SERVICE,
+                PROXY_HOST,
+                PROXY_PORT,
+                HTTP_PROXY_USERNAME,
+                HTTP_PROXY_PASSWORD,

```

Comment:
Instead of adding the individual properties, it would be cleaner to just use `ProxyConfiguration.createProxyConfigPropertyDescriptor()`.

Reply:
I did it like It was done [here](https://github.com/apache/nifi/blob/master/nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/AbstractGCPProcessor.java#L123). So I should remove the individual properties and just let the `ProxyConfiguration...` thing ?

Reply:
Thanks for the link, but yes, removing the individual property descriptors and just using the configuration service simplifies the changes. If there is some other use case that necessitates individual properties, that could be considered separately, but starting with just the proxy configuration service property avoids introducing more properties.